json-logic/example.json

```diff
@@ -0,0 +1,318 @@
+{
+  "meta": {
+    "author": "Alexander",
+    "languages": [
+      "DE"
+    ],
+    "name": "Example",
+    "version": "0.1"
+  },
+  "questions": [
+    {
+      "id": "q1_contact",
+      "text": "Gab es Kontakt zu bestätigten Fällen?",
+      "type": "Boolean"
+    },
+    {
+      "id": "q2_contact_when",
+      "text": "Wann trat der Kontakt auf?",
+      "type": "Date",
+      "skipIf": {
+        "!": {
+          "var": "q1_contact.value"
+        }
+      }
+    },
+    {
+      "id": "q3_symptoms",
+      "text": "Symptome?",
+      "type": "Multiselect",
```

Comment:
I am not a fan of the multi-select. Why?
 - Makes it harder to use for data evaluation purposes
 - Makes it harder for an application not based on a visual interface, e.g. telephone call, because they have to do extra parsing. 

I propose: 
Using multiple `boolean` questions, with
 -  either leaving the grouping to the application
 - adding nested groups (my favourite) (e.g. FHIR "type: group")
 - or adding a `group: "symptoms"` property for grouping (my least favourite)

Reply:
The proposed solutions complicate the format and make it harder for **visual interfaces which are the main use-case**. 

**Data evaluation** is rather related to the **answer format** not to the definition format of the questionnaire.

I'd opt for no, but I'm open for additional input.

Reply:
Than we have to decide and change one of them: 
> "As platform-independend as possible" 

is not 

> "visual interfaces which are the main use-case." 

Of course 
> Data evaluation is rather related to the answer format not to the definition format of the questionnaire.

but it requires additional transformation, which makes it harder. 

I just wanted to inform about the decision. The grouping of  questions was something we also discussed in one of the calls, so this might be a use case for them.

Reply:
I guess, the first and last approach won't make it harder for the visual interface (only nested groupds would require additional handling). 

If the list of multi-select values gets longer, multi-selects could cause the user to miss values (due to lack of attention). If the user is required to explicitly select yes / no for every question, we might avoid that?

Reply:
Ok, I see the points. @DanielHabenicht could you please show how your proposal would look like for the the sympthoms question in our example questionaire?

Please include the evaluation part as well.

Reply:
Update from @alexanderthieme - Multiselects are required. I think we should try out the proposed alternatives on an example and then we see which works best.

Reply:
Here you go: https://gist.github.com/DanielHabenicht/b4fe8f6085fbec3361695f92e48b7eb9

Reply:
There is nothing keeping the Fronted from rendering a group as a multi-select.

Reply:
Thanks!

> With multi-select thats not possible because we should not use `filter` anymore

We can still use `in`.

> Provide a custom count like `question_id.count`

Makes sense, but the other way around I could just argue that we expose a `boolean` per answer value as variable, e.g. `q1_symtpoms.option.cough.wasSelecten` (or so). Maybe we should do that.


Also, there are seperate questions (Haben sie Husen, haben sie Fieber) which makes it impossible for the UI to summarize this into a single multi-select question. 

I appreciate the thoughts put in, but I think that with all the workarounds in place using explicit yes/no questions and introducing groups makes the format less intuitive to write with little benefit.

Reply:
> Thanks!
> 
> > With multi-select thats not possible because we should not use `filter` anymore
> 
> We can still use `in`.

Didn't see that! Thx

> > Provide a custom count like `question_id.count`
> 
> Makes sense, but the other way around I could just argue that we expose a `boolean` per answer value as variable, e.g. `q1_symtpoms.option.cough.wasSelecten` (or so). Maybe we should do that.

But that's an additional implementation, that wouldn't be needed if we use groups. 

> > There is nothing keeping the Fronted from rendering a group
> 
> There are seperate questions (Haben sie Husen, haben sie Fieber) which makes it impossible for the UI to summarize this into a single multi-select question nicely. We'd need an additional text for the group (eg. Hatten Sie Symptome).

But we only need one text. A telephone questionnaire also does not know what to ask for each option. 

> I appreciate the thoughts put in, but I think that with all the workarounds in place for getting multiselect (using explicit yes/no questions and introducing groups) makes the format less intuitive to write with little benefit.